raiden-ts/src/services/epics.ts

```diff
@@ -897,17 +897,29 @@ function validateRoute$(
 }
 
 function validateRouteTargetAndEventuallyThrow(action: pathFind.request, latest: Latest): void {
-  const { tokenNetwork, target } = action.meta;
+  const { tokenNetwork, target, value } = action.meta;
   const { state, presences } = latest;
 
-  if (!Object.values(state.tokens).includes(tokenNetwork))
-    throw new RaidenError(ErrorCodes.PFS_UNKNOWN_TOKEN_NETWORK, { tokenNetwork });
+  assert(Object.values(state.tokens).includes(tokenNetwork), [
+    ErrorCodes.PFS_UNKNOWN_TOKEN_NETWORK,
+    { tokenNetwork },
+  ]);
+
+  assert(presences[target].payload.available, [ErrorCodes.PFS_TARGET_OFFLINE, { target }]);
 
-  if (!presences[target].payload.available)
-    throw new RaidenError(ErrorCodes.PFS_TARGET_OFFLINE, { target });
+  assert(!presences[target].payload.caps?.[Capabilities.NO_RECEIVE], [
+    ErrorCodes.PFS_TARGET_NO_RECEIVE,
+    { target },
+  ]);
 
-  if (presences[target].payload.caps?.[Capabilities.NO_RECEIVE])
-    throw new RaidenError(ErrorCodes.PFS_TARGET_NO_RECEIVE, { target });
+  assert(
+    Object.values(state.channels).some(
+      (channel) =>
+        channelCanRoute(state, presences, tokenNetwork, channel.partner.address, target, value) ===
```

Comment:
I think we should rethink the `target` parameter for this function in general. I forgot the details of this function and while reading this I was slightly confused. So actually the target is only used to check if the partner is the target or not and if not check if he can mediate. Rather I would use a boolean flag that defines if the partner needs to mediate.
I would eventually open a tiny issue for that if you agree with me that this could be improved.

Reply:
Done with #1968